src/State/Provider/DeserializeProvider.php

```diff
@@ -114,4 +117,30 @@ public function provide(Operation $operation, array $uriVariables = [], array $c
 
         return $data;
     }
+
+    private function normalizeExpectedTypes(?array $expectedTypes = null): array
+    {
+        $normalizedTypes = [];
+
+        foreach ($expectedTypes ?? [] as $expectedType) {
+            $normalizedType = $expectedType;
+
+            if (class_exists($expectedType) || interface_exists($expectedType)) {
+                try {
+                    $normalizedType = $this->resourceMetadataCollectionFactory->create($expectedType)->getOperation()->getShortName();
+                } catch (\Throwable) {
+                    // Do nothing
+                }
+
+                if (null === $normalizedType) {
+                    $classReflection = new \ReflectionClass($expectedType);
+                    $normalizedType = $classReflection->getShortName();
+                }
```

Comment:
I think this is quite sufficient actually (I thought we could re-use the current `operation` in the normalizers/denormalizers) but indeed in that case the type may be another one then the current operation's one...

Reply:
Yes, we should fetch the operation exactly for class in `$expectedType`.